src/dune_rules/coq_rules.ml

```diff
@@ -96,6 +96,13 @@ let select_native_mode ~sctx ~(buildable : Buildable.t) =
   else
     snd buildable.mode
 
+let rec resolve_first lib_db = function
+  | [] -> assert false
```

Comment:
You could also use `Error (User_error.E (User_error.make ...))` to avoid the assert false here, as this function already returns a `Result.t`.

Reply:
Sorry, what (user) error would you report? This case only happens when calling the function with an empty list, this just doesn't make any sense to me.